pkg/blockbuilder/tsdb.go

```diff
@@ -100,6 +101,44 @@ func (b *TSDBBuilder) Process(ctx context.Context, rec *kgo.Record, lastBlockMax
 		}
 	}()
 
+	mapAppendError := func(err error) error {
```

Comment:
maybe move this to a function of its own so that Process can be read a bit more quickly?